src/snippets/handling-null.md

```diff
@@ -0,0 +1,66 @@
+---
+description: Using and Handling Null snippet
+---
+
+# Handling Null
+
+Handling Null as a union type for optional return values, and simulating try/catch for errors in AssemblyScript
+
+```editor
+#!optimize=size&runtime=full
+class MyValue {
+  value: i32;
+
+  constructor(value: i32) {
+    this.value = value;
+  }
+}
+
+// Using a class type here, as some types are not nullable
+function getMyValue(isAble: boolean): MyValue | null {
+  let myValue = new MyValue(24);
+  if (isAble) {
+    return myValue;
+  } else {
+    return null;
+  }
+}
+
+export function positiveAddWithMyValue(a: i32): i32 {
+  let myValue = getMyValue(a > 0);
+  if (myValue == null) {
```

Comment:
This is more of a comment than an actual suggestion, since writing it this way is typical, even though it behaves a little different from `===`. It does not in this case, because `MyValue` does not declare an `==` overload, but if it would, `===` (identity equals) can be precomputed while `==` (calls overload function) cannot. Only relevant if eliminating branches with constant values is the goal, and a class with such an overload is the subject (i.e. a `string`). Thought I mention in case you have an idea how to convey this in a way that it doesn't become "too much" for a snippet.

Reply:
Ah yeah so I don't know a good way to express that in a snippet. Perhaps we can leave a comment (in markdown) as a general idea? 

Ideally, we would have a double equals vs. triple equals snippet that shows off this behavior and link to it from this example :smile: